mitodl/mitxonline
# Update Wagtail to v5

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/pull/1527

#### What's this PR do?
Updates Wagtail to v5.

#### How should this be manually tested?

1. Run the quick start command described here: https://github.com/mitodl/mitxonline/blob/main/docs/source/configuration/quickstart.rst
2. Run `docker-compose run web ./manage.py migrate`
3. Run `docker-compose run web ./manage.py update_exchange_rates`
4. Create a flexible price page through http://mitxonline.odl.local:8013/cms
5. Create a signature page through http://mitxonline.odl.local:8013/cms
6. Create a certificate page through http://mitxonline.odl.local:8013/cms and link it to a course.
7. Go to the course page in the CMS (http://mitxonline.odl.local:8013/cms) for the course used in step 4 and link the flexible pricing page created during step 3.
8. Update the course run related to the course used in step 4 (http://mitxonline.odl.local:8013/admin/courses/courserun/) to have a start date in the past, end date in the future, and enrollment start date in the past.
9. Add the course as a dependency to the existing Program at http://mitxonline.odl.local:8013/admin/courses/program/
10. Login to MITx Online using a test learner.
11. Enroll in the free version of the course you've been referencing.
12. Ensure that the required fields at http://mitxonline.odl.local:8013/profile/ are populated.
13. Apply for flexible pricing for the course.  I used a low income level ($5 USD) in order to be redirected to the "pending" flexible price page.
14. Verify that you see the expected flexible pricing status screen after applying.  Leave this page open while you complete the next step.
15. Open the staff-dashboard (http://mitxonline.odl.local:8013/staff-dashboard/) and approve your flexible pricing request.
16. Refresh the page from step 11 and verify that you see the approved flexible pricing screen.
17. Go to the MITx Online dashboard and navigate to the course details page for the course you are enrolled in.
18. Purchase the certificate track for the course.
19. Create a new course run grade record for your user and that course with a passing grade(http://mitxonline.odl.local:8013/admin/courses/courserungrade/).
20. Create a new course run certificate record for your user and that course (http://mitxonline.odl.local:8013/admin/courses/courseruncertificate/).
21. Navigate to the dashboard and click on the "View certificate" link on the course card.
22. Verify that the course certificate is displayed correctly.
23. Go back to the CMS (http://mitxonline.odl.local:8013/cms/) and create a program certificate for the existing program.
24. Create a program certificate record for your user at http://mitxonline.odl.local:8013/admin/courses/programcertificate/
25. Go to the MITx Online dashboard, click the Programs tab, and click "View program certificate".
26. Verify that the program certificate is displayed correctly.

#### Any background context you want to provide?

- I ran into a lot of migration issues that were attributed to https://github.com/mitodl/mitxonline/blob/e10fc8d028528d9acd8a0d9b837aed4e68a9d696/cms/migrations/0026_certificate_index_page_slug.py. Commenting out this migration's operation resolved these issues. Nathan and I don't believe there will be a problem with having this operation commented out since it has been run in production.  My suspicion is that this troublesome migration lead to dependencies with other MITx Online models which have Wagtail dependencies themselves but are migrated after the troublesome migration.
- There was a bug in v4 that impacting us.  This was fixed in v5 which is why I went all the way to v5.
- Wagtail release notes and upgrade notes: https://docs.wagtail.org/en/stable/releases/upgrading.html
